plugins/api/src/atomic_apply_acl_operations.cpp

```diff
@@ -378,7 +379,11 @@ namespace
             throw_if_invalid_acl(acl);
 
             log::api::trace("Retrieving entity ID ...");
-            const auto entity_id = get_entity_id(_db_conn, _op.at("entity_name").get<std::string>());
+            auto zone = std::string{getLocalZoneName()};
+            if(_op.find("zone_name") != std::end(_op)) {
+                zone = _op.at("zone_name").get<std::string>();
+            }
```

Comment:
Perhaps this can be done without allocating memory?

Reply:
I think switching to `std::string_view` should work here-- it has a constructor for `char *` and shouldn't allocate any more memory.

Reply:
I'm nervous about replacing std::string with string_view on the `.get` call. The nlohmann JSON library offers `.get_ref` for situations where the data is readonly.

What do you think about this?
```cpp
std::string_view zone = getLocalZoneName();
if (const auto iter = _op.find("zone_name"); iter != std::end(_op)) {
    zone = iter->get_ref<const std::string&>();
}
```

Reply:
Seems good. I'll add it!